api/users/serializers.py

```diff
@@ -373,6 +379,7 @@ class UserSettingsSerializer(JSONAPISerializer):
     two_factor_enabled = ser.SerializerMethodField()
     subscribe_osf_general_email = ser.SerializerMethodField()
     subscribe_osf_help_email = ser.SerializerMethodField()
+    secret = ser.SerializerMethodField(read_only=True)
```

Comment:
I am actually going to need something else, I think. I don't see a way of knowing whether or not two_factor is validated, other than assuming that, because two_factor_enabled is `true` and `secret` is `null`, it's not yet validated. Maybe we should go ahead and have a read-only `two_factor_confirmed` field just so it's explicit from the API.